src/main/java/laura/TaskList.java

```diff
@@ -0,0 +1,192 @@
+            }
+        } catch (IOException e) {
+            UI.send(e.getMessage());
+        }
+    }
+
+    /**
+     * Takes a line of encoded data, decodes it,
+     * and returns the corresponding Task
+     *
+     * @param data The encoded data
+     * @return The decoded Task
+     * @throws LauraException When the data is not encoded in a valid format
+     */
+    private Task decode(String data) throws LauraException {
+        String[] args = data.split(Pattern.quote("|"));
+        if (args.length < 2 || !"TDE".contains(args[0]) || !"01".contains(args[1])) {
+            throw new DecodeException();
+        }
+        boolean isDone = args[1].equals("1");
+
+        return switch (args[0]) {
+            case "T" -> {
+                if (args.length != 3) {
+                    throw new DecodeException();
+                }
+                yield new ToDoTask(isDone, args[2]);
+            }
+            case "D" -> {
+                if (args.length != 4) {
+                    throw new DecodeException();
+                }
+                yield new DeadlineTask(isDone, args[2], args[3]);
+            }
+            case "E" -> {
+                if (args.length != 5) {
+                    throw new DecodeException();
+                }
+                yield new EventTask(isDone, args[2], args[3], args[4]);
+            }
+            default -> throw new DecodeException();
+        };
+    }
+
+    /**
+     * Load the Tasks saved from local into the TaskList
+     */
+    public void load() {
+        File file = new File(this.dataPath);
+        try {
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String data = scanner.nextLine();
+                try {
+                    this.list.add(this.decode(data));
+                } catch (LauraException e) {
+                    UI.send(e.getMessage());
+                }
+            }
+        } catch (FileNotFoundException e) {
+            UI.send(e.getMessage());
+        }
+
+    }
+
+    /**
+     * Add a Task to the TaskList
+     * @param task The task to be added
+     */
+    public void add(Task task) {
+        this.list.add(task);
+        this.save();
+        UI.send("Got it! I've added this task:\n" + task);
+    }
+
+    /**
+     * Delete a Task from the TaskList
+     * @param index The index of the Task in the TaskList to be deleted
+     * @throws LauraException If there is no corresponding Task for the index given
+     */
+    public void delete(int index) throws LauraException {
+        if (index > this.list.size() || index < 1) {
+            throw new LauraException("Sorry, that task does not exist!");
+        }
+        Task removed = this.list.remove(index - 1);
+        this.save();
+        UI.send("Noted. I've removed this task:\n"
+                + removed +
+                "\nNow you have " + this.list.size() + " in this list.");
```

Comment:
Perhaps the operator should be part of the line break here?
```suggestion
        UI.send("Noted. I've removed this task:\n"
                + removed 
                + "\nNow you have " + this.list.size() + " in this list.");
```